app/src/androidTest/java/org/mozilla/fenix/ui/DownloadTest.kt

```diff
@@ -193,30 +196,40 @@ class DownloadTest {
     @SmokeTest
     @Test
     fun openPDFInBrowserTest() {
-        navigationToolbar {
-        }.enterURLAndEnterToBrowser(downloadTestPage.toUri()) {
-            clickLinkMatchingText(pdfFileName)
-            verifyUrl(pdfFileURL)
-            verifyPageContent(pdfFileContent)
+        runWithCondition(
+            // Returns the GeckoView channel set for the current version, if a feature is limited to Nightly.
+            activityTestRule.activity.components.core.engine.version.releaseChannel == EngineReleaseChannel.NIGHTLY,
```

Comment:
I would add a comment to unwrap this once the feature graduates out of Nightly. I wonder if we should also feature detect (there could be cases where a feature maybe gets backed out or turned off or something).

Reply:
That would imply having access to GeckoView prefs I suppose? and that is yet not possible either.